event_sourcery/repository.py

```diff
@@ -19,27 +18,24 @@ def __init__(
     class NotFound(Exception):
         pass
 
-    @contextmanager
-    def new(self, stream_id: StreamId) -> Iterator[TAggregate]:
-        changes: list[Event] = []  # To be mutated inside aggregate
-        aggregate = self._aggregate_cls()
-        aggregate.__restore_aggregate_state__(
-            past_events=[], changes=changes, stream_id=stream_id
-        )
-        yield aggregate
-        self._event_store.append(
-            stream_id=stream_id, events=changes, expected_version=0  # new aggregate
-        )
+    def load(self, stream_id: StreamId) -> TAggregate:
+        aggregate = object.__new__(self._aggregate_cls)
```

Comment:
I don't buy (for now) passing in an `Aggregate` instance.

That would also enforce a certain style on library's user and prevent patterns like
https://github.com/peterpodgorski/python_event_sourcing_cqrs/blob/main/domain.py#L72

Even though I am not sure if I would personally followed that, I don't want to take that option away.

Me, personally, would use a factory `@classmethod` and used `__init__` to initialise fields

Reply:
my preferred style
```python
class LightSwitch(Aggregate):
    def __init__(self) -> None:
        super().__init__()
        self._shines = False
        
    @classmethod
    def enabled(cls) -> "LightSwitch":
        instance = LightSwitch()
        instance._event(TurnedOn)
        return instance
```

(some) others' preferred style

```python
class LightSwitch(Aggregate):
    def __init__(self, turned_on: bool) -> None:
        super().__init__()
        self._shines = False
        
        if turned_on:
            self._event(TurnedOn)
```